controllers/front/OrderController.php

```diff
@@ -145,7 +148,12 @@ private function restorePersistedData(CheckoutProcess $process)
         if (!is_array($data)) {
             $data = [];
         }
-        $process->restorePersistedData($data);
+
+        $cartChecksum = new CartChecksum();
+        $checksum = $cartChecksum->checksum($this->context->cart);
```

Comment:
I like the idea to let the cart checksum generator to another class, so the single job of this class is to get the checksum.

**BUT!**

If the cart is the only element needing a checksum, I'm okay with your idea @mickaelandrieu.

Else, if it is not the only element, we need:

  - A checksum interface
  - The different implementations